### PR TITLE
Refreshing fossilise buttons when a new fossil is created

### LINKED ISSUE #3973: Fossilise buttons should refresh when a new fossil is created
fossilise buttons should refresh when hitting ok on the fossilise dialog so that buttons for the species that just got fossilised would get marked as fossilised to make it more clear to the player what happened, and they'd probably learn what the already fossilised buttons look like.

### PR DESCRIPTION
**Pull request by dligr**
**Brief Description of What This PR Does**

This pull request makes fossilise buttons refresh after fossilising a new species. So, if the player fossilises a species that wasn't fossilised before, the buttons of this species will change their texture to the texture of already fossilised buttons to reflect this change.

**Related Issues**

Fixes #3973

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/general/base_stage/CreatureStageHUDBase.cs [hunk 1]
```diff
@@ -432,6 +432,9 @@ public override void _Ready()
         fossilisationButtonLayer = GetNode<Control>(FossilisationButtonLayerPath);
         fossilisationDialog = GetNode<FossilisationDialog>(FossilisationDialogPath);
 
+        fossilisationDialog.Connect(nameof(FossilisationDialog.OnSpeciesFossilised), this,
```
Comment (hhyyrylainen) [inline src/general/base_stage/CreatureStageHUDBase.cs:435]: Signal connections should be done with the Godot editor.

File: src/general/base_stage/CreatureStageHUDBase.cs [hunk 2]
```diff
@@ -676,6 +679,21 @@ public override void PauseButtonPressed(bool buttonState)
...
+    /// </summary>
+    public void RefreshFossilisationButtonsFossilStatus()
+    {
+        var fossils = FossilisedSpecies.CreateListOfFossils(false);
+
+        foreach (FossilisationButton button in fossilisationButtonLayer.GetChildren())
+        {
+            var species = button.AttachedEntity.Get<SpeciesMember>().Species;
+            var fossilised = FossilisedSpecies.IsSpeciesAlreadyFossilised(species.FormattedName, fossils);
+            button.AlreadyFossilised = fossilised;
```
Comment (hhyyrylainen) [inline src/general/base_stage/CreatureStageHUDBase.cs:693]: This doesn't update the tooltip... Compare what happens in `MicrobeHUD.ShowFossilisationButtons`. I think it's probably better to just have a `abstract void UpdateFossilisationButtonStates` in this class and let the specialized HUD classes handle updating the fossilization buttons as they know the entity types etc they should check. And that also allows sharing the common code for setting up the fossilization button between this method and `ShowFossilisationButtons`.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Thank you for contributing to Thrive. The overall approach this PR takes looks good, but I have 2 code architecture and style improvement suggestions.

File: src/late_multicellular_stage/MulticellularStage.tscn [hunk 3]
```diff
@@ -103,4 +103,5 @@ __meta__ = {
 [connection signal="OnOpenInventoryPressed" from="MulticellularHUD" to="PlayerMulticellularInput" method="OpenInventory"]
 [connection signal="OnOpenMenu" from="MulticellularHUD" to="PauseMenu" method="Open"]
 [connection signal="OnOpenMenuToHelp" from="MulticellularHUD" to="PauseMenu" method="OpenToHelp"]
+[connection signal="OnSpeciesFossilised" from="MulticellularHUD/FossilisationDialog" to="." method="UpdateFossilisationButtonStates"]
```
Comment (hhyyrylainen) [inline src/late_multicellular_stage/MulticellularStage.tscn:106]: Is this connected correctly? The target seems to be the stage and not the HUD.

Comment (dligr) [reply src/late_multicellular_stage/MulticellularStage.tscn]: Right, should be fixed now

**Review by hhyyrylainen — APPROVED** [review]
Code changes look fine now. I'll try to get someone to playtest this next before merging.

**Comment by hhyyrylainen** [comment]
I tried to playtest this but I got the following error and the buttons didn't update: `ERROR: In Object of type 'Control': Attempt to connect nonexistent signal 'OnSpeciesFossilised' to method 'Control.UpdateFossilisationButtonStates'.
`

**Comment by dligr** [comment]
> I tried to playtest this but I got the following error and the buttons didn't update: `ERROR: In Object of type 'Control': Attempt to connect nonexistent signal 'OnSpeciesFossilised' to method 'Control.UpdateFossilisationButtonStates'. `

I was able to reproduce this issue with the same error by reverting the changes made to `FossilisationDialog.cs`. Could the issue be that this file isn't updated on your end?

**Comment by hhyyrylainen** [comment]
That's weird, I just checked the PR again and now I get the expected result. I tried pulling on the PR branch but git said it was up to date. So this was some weird situation where I earlier had the wrong commit when doing the exact same thing that worked now a bit later.

**Pull request merged**